internal/controller/static_nodepool_controller.go

```diff
@@ -0,0 +1,117 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	ConfigMapName = "static-nodepools-config"
+)
+
+// StaticNodepoolReconciler reconciles static nodepools based on a ConfigMap.
+type StaticNodepoolReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+
+	Provider                    cloud.Provider
+	Concurrency                 int
+	StaticNodepoolCreateTimeout time.Duration
+}
+
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
+
+func (r *StaticNodepoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	lg := ctrllog.FromContext(ctx)
+
+	// Only reconcile if this configmap is the one we are looking for.
+	if req.Name != ConfigMapName || req.Namespace != "default" {
+		return ctrl.Result{}, nil
+	}
+
+	lg.V(3).Info("Reconciling static nodepools")
+
+	var cm corev1.ConfigMap
+	if err := r.Get(ctx, req.NamespacedName, &cm); err != nil {
+		if apierrors.IsNotFound(err) {
+			lg.Info("Static nodepools config map not found. Skipping reconciliation.", "configmap", req.NamespacedName.String())
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get configmap %s: %w", req.NamespacedName.String(), err)
+	}
+
+	reservationsYAML, ok := cm.Data["reservations"]
+	if !ok {
+		lg.Info("No 'reservations' key in configmap. Skipping reconciliation.", "configmap", req.NamespacedName.String())
+		return ctrl.Result{}, nil
+	}
+
+	var reservationNames []string
+	if err := yaml.Unmarshal([]byte(reservationsYAML), &reservationNames); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to unmarshal reservations from configmap %s: %w", req.NamespacedName.String(), err)
+	}
+
+	nodepoolConfigYAML, ok := cm.Data["nodepoolConfig"]
+	if !ok {
+		lg.Info("No 'nodepoolConfig' key in configmap. Skipping reconciliation.", "configmap", req.NamespacedName.String())
+		return ctrl.Result{}, nil
+	}
+
+	var nodepoolConfig cloud.NodePoolConfig
+	if err := yaml.Unmarshal([]byte(nodepoolConfigYAML), &nodepoolConfig); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to unmarshal nodepoolConfig from configmap %s: %w", req.NamespacedName.String(), err)
+	}
+
+	reconcileCtx := cloud.WithStaticNodePoolCreateTimeout(ctx, r.StaticNodepoolCreateTimeout)
+	reconcileCtx = cloud.WithStaticNodePoolConcurrency(reconcileCtx, r.Concurrency)
+
+	for _, reservationName := range reservationNames {
+		lg.Info(fmt.Sprintf("Ensuring static nodepool for reservation: %s", reservationName))
+		if err := r.Provider.EnsureStaticNodePools(reconcileCtx, reservationName, &nodepoolConfig); err != nil {
+			if apierrors.IsNotFound(err) {
+				lg.Info(fmt.Sprintf("Reservation %s not found, will retry later: %v", reservationName, err))
```

Comment:
Why is this logged separately? Wouldnt the reason of not-found just be logged naturally?